src/child.c

```diff
@@ -92,7 +93,20 @@ static void sighandler (int signal)
         case SIGPIPE:
             operlog("Got SIGPIPE, ignored");
             break;
-    }
+        case SIGSEGV:
+            operlog("Segmentation fault");
+            // from https://stackoverflow.com/questions/77005/how-to-automatically-generate-a-stacktrace-when-my-program-crashes
+            void *array[10];
+            size_t size;
+
+            // get void*'s for all entries on the stack
+            size = backtrace(array, 10);
+
+            // print out all the frames to stderr
+            fprintf(stderr, "Segmentation fault:\n");
+            backtrace_symbols_fd(array, size, STDERR_FILENO);
+            exit(1);
```

Comment:
Can you instead raise(SIGSEGV); so that we still exit on that signal and have a core dump ? You'll also need to set SA_RESETHAND on the sigaction struct for SIGSEGV. This will allow the signal handler to return to its default after being triggered, preventing a loop.